File: data/code_pairs/fortran/1819.f90
SUBROUTINE loop12_F77(N, y, x, a, b, c, d, u)
      INTEGER N, i
      REAL*8 y(N), x(N), a(N), b(N), c(N), d(N), u

      DO i=1,N
          y(i) = u + a(i)
          x(i) = a(i) + b(i) + c(i) + d(i)
      END DO
      RETURN
END

PROGRAM testLoop12_F77
    IMPLICIT NONE
    INTEGER, PARAMETER :: N = 5
    REAL*8 y(N), x(N), a(N), b(N), c(N), d(N), u
    INTEGER i

    ! Initialize test data
    u = 2.0
    DO i = 1, N
        a(i) = i * 1.0
        b(i) = i * 2.0
        c(i) = i * 3.0
        d(i) = i * 4.0
    END DO

    ! Call the subroutine
    CALL loop12_F77(N, y, x, a, b, c, d, u)

    ! Output results for demonstration
    DO i = 1, N
        PRINT *, "y(", i, ") = ", y(i), ", x(", i, ") = ", x(i)
    END DO
END PROGRAM testLoop12_F77

File: data/code_pairs/cpp/1819.cpp
#include <vector>
#include <iostream>

void loop12_F77(int N, std::vector<double>& y, std::vector<double>& x, 
                const std::vector<double>& a, const std::vector<double>& b, 
                const std::vector<double>& c, const std::vector<double>& d, double u) {
    for (int i = 0; i < N; ++i) {
        y[i] = u + a[i];
        x[i] = a[i] + b[i] + c[i] + d[i];
    }
}

// Main function to demonstrate the loop12_F77 functionality
int main() {
    const int N = 5;
    std::vector<double> y(N), x(N), a(N), b(N), c(N), d(N);
    double u = 2.0;

    // Initialize test data
    for (int i = 0; i < N; ++i) {
        a[i] = (i + 1) * 1.0;
        b[i] = (i + 1) * 2.0;
        c[i] = (i + 1) * 3.0;
        d[i] = (i + 1) * 4.0;
    }

    // Call the function
    loop12_F77(N, y, x, a, b, c, d, u);

    // Output results for demonstration
    for (int i = 0; i < N; ++i) {
        std::cout << "y(" << i + 1 << ") = " << y[i] << ", x(" << i + 1 << ") = " << x[i] << std::endl;
    }

    return 0;
}